### ISSUE TITLE
using network drive on windows as media path

### ISSUE THREAD

**Issue by bigretromike** [miyachan/torako#15]
For anyone having issue with "other io error" when using mounted smb share as drive `media_path`.

short:
set both `media_path` and `tmp_path` on same drive

long:
if you compile torako with newer rust (one including LLVM 13.0) you will get end up with newer error message: `the system cannot move the file to a different disk drive`. This one indicate that function used to move files from `tmp` to `media` use function that rename path and not move files around and you cannot rename file to another drive.

**Comment by miyachan** [comment]
Issue is here: https://github.com/miyachan/torako/blob/1234fa0d7b8bffd8da1bcd5a0e071ead0c48d642/src/storage/asagi/storage/filesystem.rs#L196

[`tokio::fs::rename`](https://docs.rs/tokio/0.1.16/tokio/fs/fn.rename.html) doesn't support renaming across mount points. For best performance, they _should_ be on the same drive.

**Comment by bigretromike** [comment]
@miyachan you are correct. I wanted to post this here, just in case someone was also struggling with it, because that information wasn't available when I was searching for it :-)

**Comment by bigretromike** [comment]
Also using same but slow drive (or slow comparing to local drive) torako - to be more clear `tokio` false postivie drive error not being the same drive when in real life its a timeout issue. For temporary workaround one can store tmp/media on local drive and push `media` to `nas/server` via schedule
Also limiting `concurrent_downloads` lower the count of `lost` files in `tmp`
